modules/openapi-json-schema-generator/src/main/resources/python/rest.handlebars

```diff
@@ -125,6 +128,11 @@ class RESTClientObject(object):
                   len(timeout) == 2):
                 timeout = urllib3.Timeout(connect=timeout[0], read=timeout[1])
 
+        {{#if hasAwsSignatureV4Methods}}
+        if self.aws_sigv4_info:
+            headers.extend(self.aws_sigv4_info.get_aws_signature_v4_headers(method, url, body))
```

Comment:
This code should be route specific.
How about moving this code into update_params_for_auth
And moving the invocation of that call after the header extension?
https://github.com/openapi-json-schema-tools/openapi-json-schema-generator/blob/master/modules/openapi-json-schema-generator/src/main/resources/python/api_client.handlebars#L1063

Reply:
Sorry was taking a break all weekend so just getting back to this! 

I like the idea of it being per route, however there is a comment above where used_headers.update is calling saying it should be called after any auth settings. Does that not apply anymore?

Reply:
No worries. Also I am on vacation so forgive some delayed responses from me here. It kind of doesn't apply. I wrote it. The default headers must come first. Update has to happen after, and those are spec defined headers. For your signing and the other security scheme they must come last because they depend upon the payload in general, right?
For the other security schemes, the Autherization header is disallowed as an explicit header input. So it is probably fine to update all security info last.